cli/src/commands/describe.rs

```diff
@@ -65,39 +77,159 @@ pub(crate) fn cmd_describe(
-    }
+    let tx_description = if commits.len() == 1 {
+        format!("describe commit {}", commits[0].id().hex())
+    } else {
+        format!(
+            "describe commit {} and {} more",
+            commits[0].id().hex(),
+            commits.len() - 1
+        )
+    };
 
-    let description = if args.stdin {
+    let shared_description = if args.stdin {
         let mut buffer = String::new();
         io::stdin().read_to_string(&mut buffer)?;
-        buffer
+        Some(buffer)
     } else if !args.message_paragraphs.is_empty() {
-        join_message_paragraphs(&args.message_paragraphs)
-    } else if args.no_edit {
-        commit.description().to_owned()
+        Some(join_message_paragraphs(&args.message_paragraphs))
     } else {
-        if commit_builder.description().is_empty() {
-            commit_builder.set_description(command.settings().default_description());
+        None
+    };
+
+    let commit_descriptions: Vec<(_, _)> = if args.no_edit || shared_description.is_some() {
+        commits
+            .iter()
+            .map(|commit| {
+                let new_description = shared_description
+                    .as_deref()
+                    .unwrap_or_else(|| commit.description());
+                (commit, new_description.to_owned())
+            })
+            .collect()
+    } else {
+        let repo = tx.base_repo().clone();
+        let temp_commits: Vec<(_, _)> = commits
+            .iter()
+            // Edit descriptions in topological order
+            .rev()
+            .map(|commit| -> Result<_, CommandError> {
+                let mut commit_builder = tx
+                    .mut_repo()
+                    .rewrite_commit(command.settings(), commit)
+                    .detach();
+                if commit_builder.description().is_empty() {
+                    commit_builder.set_description(command.settings().default_description());
+                }
+                if args.reset_author {
+                    let new_author = commit_builder.committer().clone();
+                    commit_builder.set_author(new_author);
+                }
+                let temp_commit = commit_builder.write_hidden()?;
+                Ok((commit.id(), temp_commit))
+            })
+            .try_collect()?;
+
+        if temp_commits.len() == 1 {
+            let (_, temp_commit) = &temp_commits[0];
+            let template = description_template(&tx, "", temp_commit)?;
+            let description = edit_description(&repo, &template, command.settings())?;
+
+            vec![(&commits[0], description)]
+        } else {
+            let ParsedBulkEditMessage {
+                descriptions,
+                missing,
+                duplicates,
+                unexpected,
+            } = edit_multiple_descriptions(&mut tx, &repo, &temp_commits, command.settings())?;
+            if !missing.is_empty() {
+                return Err(user_error(format!(
```

Comment:
(Not really for this PR): I wonder if we should add a TODO to somehow make sure the user doesn't necessarily lost all their work here.

One option is to give the user the filename of their work and a command to continue editing from there. (Or, perhaps, it'd be nicer to give them a new (syntactically correct) template and put their old work into `JJ:` comments.)

Another option would be to give the user a menu where they have to choose between aborting or continuing their work (probably again with their old work commented out).

Reply:
I think we can probably open a separate issue for this.